src/ytdl_sub/utils/file_handler.py

```diff
@@ -353,8 +353,23 @@ def move(cls, src_file_path: Union[str, Path], dst_file_path: Union[str, Path]):
             Source file
         dst_file_path
             Destination file
+
+        Raises
+        ------
+        OSError
+            Cross-device link workaround
         """
-        shutil.move(src=src_file_path, dst=dst_file_path)
+        try:
+            shutil.move(src=src_file_path, dst=dst_file_path)
+        except OSError as os_error_exc:
+            # Invalid cross-device link
+            # Can happen from using os.rename under the hood, which requires the two file on the
+            # same filesystem. Work around it by copying and deleting the file
+            if os_error_exc.errno == 18:
```

Comment:
Out of curiosity, why limit this to a specific errno here?

I've got a situation (will make a separate issue if desired) where I both have a file on a separate filesystem (so it gets errno 18 as expected) but *also* a permissions change (which seems to throw errno 1 instead, even though the copy works). I was able to work around it in my own install by accepting both errno 1 and errno 18 here, and it seems to work fine. Would it make more sense to just attempt the copy+delete regardless?

Reply:
I'm okay with adding support for any errno. To your point, worst case if copy and delete fails, it will just throw another exception. I'd be happy to merge a PR with this change ☺️